### PR TITLE
npm run web bugfix

### PR DESCRIPTION
**Pull request by david8611**
This edit to AppEntry.js in the node_modules should fix the following ```npm run web``` compilation error below. The problem was related to the registerRootComponent taking an Object type as parameter instead of a class. This was fixed by creating and a class (that returns the component ```App```) which extends the React.Component SuperClass. More information can be found here: https://docs.expo.io/versions/latest/sdk/register-root-component/
Skärmklipp

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,7 @@
 node_modules/**/*
+!/node_modules/expo
+/node_modules/expo/*
+!/node_modules/expo/AppEntry.js
```
Comment (david8611) [inline .gitignore:4]: This was added to gitignore to exclude everything in ```/node_modules``` and only include the file ```AppEntry.js``` in ```/node_modules/expo/AppEntry.js```

File: node_modules/expo/AppEntry.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import React from 'react';
+
+
+import App from '../../App/';
+
+class Apps extends React.Component {
+    render() {
+      return <App/>;
+    }
+  }
```
Comment (david8611) [inline node_modules/expo/AppEntry.js:11]: This should fix the problem of App being treated as an Object instead of a Class.

File: .gitignore (hunk 1 above)
Comment (sanlil) [reply .gitignore:4]: This looks odd to me. We should not have to include anything from the node_modules folder

**Comment by sanlil** [comment]
I found a [Stack Overflow question](https://stackoverflow.com/questions/65657557/basic-app-not-working-in-web-after-changing-app-js-to-app-index-js) with a very similar setup as ours. They have (just like us) removed the `App.js` file and exchanged it for `App/index.js` which then caused this issue. The [solution](https://stackoverflow.com/a/65657865/7216176) was to add `registerRootComponent(Main)` to `index.js` and to change the main path in `package.json` from `"main": "node_modules/expo/AppEntry.js"` to `"main": "App/index.js"`

In other words, very similar to your solution @david8611 but with the advantage that we don't need to include anything from `node_modules`.

**Comment by david8611** [comment]
> I found a [Stack Overflow question](https://stackoverflow.com/questions/65657557/basic-app-not-working-in-web-after-changing-app-js-to-app-index-js) with a very similar setup as ours. They have (just like us) removed the `App.js` file and exchanged it for `App/index.js` which then caused this issue. The [solution](https://stackoverflow.com/a/65657865/7216176) was to add `registerRootComponent(Main)` to `index.js` and to change the main path in `package.json` from `"main": "node_modules/expo/AppEntry.js"` to `"main": "App/index.js"`
> 
> In other words, very similar to your solution @david8611 but with the advantage that we don't need to include anything from `node_modules`.

Neat solution. I will make the adjustment tomorrow.

**Comment by sanlil** [comment]
Looks like there are some commits from master included in this PR. Try to rebase your branch on the latest version of master again. If that doesn't work you can `git cherry-pick` the commits that you want onto a new branch or do a merge instead of rebase.

File: App/index.js [hunk 3]
```diff
@@ -1,4 +1,17 @@
...
-export default () => <Authors/>
+export default function Main() {
+  return (
+    <PaperProvider>
+      <Authors/>
+    </PaperProvider>
+  );
+}
+
+AppRegistry.registerComponent(name, () => Main);
```
Comment (sanlil) [inline App/index.js:16]: Do we need this line as well or is `registerRootComponent` enough?

**Comment by david8611** [comment]
This pull request is solved in #23.
